### ISSUE TITLE
Select link output is not correct

### ISSUE THREAD

**Issue by Jake-Moss** [AequilibraE/aequilibrae#442]
This is an issue to track the ongoing hunt for a correctness issue raised on group groups.

This branch has a test case for this issue https://github.com/Jake-Moss/aequilibrae/tree/select_link_bug_hunting

**Comment by Jake-Moss** [comment]
I took another look at this today and came up with this small patch that produces the correct *total* select link outputs within the `tmp` array.
```patch
diff --git a/aequilibrae/paths/results/assignment_results.py b/aequilibrae/paths/results/assignment_results.py
index 7d62241..35b5922 100644
--- a/aequilibrae/paths/results/assignment_results.py
+++ b/aequilibrae/paths/results/assignment_results.py
@@ -298,6 +298,14 @@ class AssignmentResults:
             # Link flows initialised
             link_flows = np.full((self.links, self.classes["number"]), np.nan)
             # maps link flows from the compressed graph to the uncompressed graph
+            breakpoint()
+            tmp = np.full(self.crosswalk.shape, 0.0)
+            for i, x in enumerate(self.select_link_loading[name]):
+                if x > 0.0:
+                    locs = np.arange(self.crosswalk.shape[0])[self.crosswalk == i]
+                    for id in self.graph.graph.link_id[self.graph.graph.__compressed_id__.isin(locs)]:
+                        tmp[id] += x
+            breakpoint()
             assign_link_loads(link_flows, self.select_link_loading[name], self.crosswalk, self.cores)
             for i, n in enumerate(self.classes["names"]):
                 # Directional Flows
```

On the test on the branch, the `tmp` looks like this
```python
print(*list(enumerate(tmp)), sep="\n ")
(0, 0.0)
(1, 0.0)
(2, 0.0)
(3, 500.0)
(4, 500.0)
(5, 500.0)
(6, 0.0)
(7, 0.0)
(8, 0.0)
(9, 0.0)
(10, 0.0)
(11, 500.0)
(12, 0.0)
(13, 0.0)
(14, 0.0)
(15, 0.0)
(16, 0.0)
(17, 500.0)
(18, 0.0)
(19, 0.0)
(20, 0.0)
(21, 0.0)
(22, 0.0)
(23, 0.0)
(24, 0.0)
(25, 0.0)
```
in which the index of the non-zero entries corresponds to the link ids in the desired output
![image](https://github.com/AequilibraE/aequilibrae/assets/49361082/deb5871e-7f25-4467-85ac-941b4bc36415)

**Comment by Jake-Moss** [comment]
To document my current understanding of the issue I'll add little explanation here.

The output of the select link algorithm itself is correct, the issue lies in the mapping back to the true links. The values need to be mapped back in the following manner.
The index of non-zero entries within `self.select_link_loading[name]` correspond to values within `self.crosswalk`, the index of those values is then the `__compressed_id__` of the link. Within the graph `__compressed_id__`s correspond to `link_id`s.

**Comment by Jake-Moss** [comment]
Looks like I over complicated it and rubber ducked myself to a solution. Just pushed a basic fix in  ea51b1233b65479c3317f82f2c7d9aa2816163c9

Heres the print out from pandas
```
    index  trace_a_ab  trace_a_ba  trace_a_tot
0       1         0.0         NaN          0.0
1       2         0.0         NaN          0.0
2       3         0.0       500.0        500.0
3       4       500.0         0.0        500.0
4       5         0.0       500.0        500.0
5       7         0.0         0.0          0.0
6       9         0.0         0.0          0.0
7      11         0.0       500.0        500.0
8      13         0.0         0.0          0.0
9      15         0.0         0.0          0.0
10     17         0.0       500.0        500.0
11     19         0.0         0.0          0.0
12     21         0.0         0.0          0.0
13     23         0.0         0.0          0.0
```
